### DISCUSSION TITLE
py_canoe improvement ideas

### DISCUSSION THREAD

**Discussion by chaitu-ycr** [chaitu-ycr/py_canoe#44] [Ideas]
post your comments here for py_canoe improvement.
please fork and create pull request if you have implementation code.

**Comment by iAmStefan** [comment]
Hello. An improvement idea would be to write a method that allows you to load LDF (LIN description file) or CAN DBC file into CANoe.

**Reply by chaitu-ycr** [reply]
https://chaitu-ycr.github.io/py-canoe/src_manual/#src.py_canoe.canoe.CANoe.add_database

**Comment by AhmedKotb10** [comment]
Hello, 
Is it possible to extend the implementation to handle execution of certain Test Groups/Test Cases ?
if yes, is it already planned?

**Reply by chaitu-ycr** [reply]
for now only execution of test modules, test environments is possible.
i am not sure whether test group execution possible via CANoe COM. i will explore and let u know once i have some idea.
please raise pull request if you have code for doing same.

**Reply by chaitu-ycr** [reply]
executing vtest test configuration is available in py-canoe version 26.1.0
[https://chaitu-ycr.github.io/[py-canoe](https://chaitu-ycr.github.io/py-canoe/#execute-test-configuration-test-units)/#execute-test-configuration-test-units](https://chaitu-ycr.github.io/%5Bpy-canoe%5D(https://chaitu-ycr.github.io/py-canoe/#execute-test-configuration-test-units)/#execute-test-configuration-test-units)

**Comment by Sona-Shetty** [comment]
Hello, Is it possible to start logging the trace window once the measurement is started using any of your available modules? (All in online mode)

**Reply by chaitu-ycr** [reply]
controlling trace window logging is not possible for now. but you can control logging block itself using these methods.
https://chaitu-ycr.github.io/py-canoe/src_manual/#src.py_canoe.canoe.CANoe.get_logging_blocks
https://chaitu-ycr.github.io/py-canoe/src_manual/#src.py_canoe.canoe.CANoe.start_stop_online_logging_block

**Comment by jaswanth091** [comment]
is there any possibility to run python script at the time of measurement is in running state ?
Need to open and start configuration everytime ?

**Reply by Trenboloni** [reply]
Hello,

i had the same problem. In the offiicial repo that method is not existing. So i created one. Hopefully it will be added to the official repo.

Just put the following code into the CANoe class:

class CANoe:

      def  get_running_instance(self, visible=True):
        
        try:
            self.application = None
            self.application = win32com.client.DispatchEx("CANoe.Application")
            self.ver = self.application.Version

            self.__init_canoe_application()
            self.__init_canoe_application_measurement()
            self.__init_canoe_application_simulation()
            self.__init_canoe_application_version()

            self.application_com_obj.Visible = visible
            
            self.__init_canoe_application_bus()
            self.__init_canoe_application_capl()
            self.__init_canoe_application_configuration()
            self.__init_canoe_application_environment()
            self.__init_canoe_application_networks()
            self.__init_canoe_application_system()
            self.__init_canoe_application_ui()
            self.__init_logging_collection()

            self.get_canoe_version_info()
            
            self.Measurement = self.application.Measurement.Running
            self.__log.debug(f'👉Measurement runnung: {self.Measurement}')
            #print(f'Measurement runnung: {self.Measurement}')

        except Exception as e:
            error_message = f'😡 Error opening CANoe configuration: {str(e)}'
            self.__log.error(error_message)
            raise PyCanoeException(error_message)
        


Then just create the instance and there you go :D.

**Reply by chaitu-ycr** [reply]
https://chaitu-ycr.github.io/py-canoe/src_manual/#src.py_canoe.canoe.CANoe.attach_to_active_application

**Comment by TuTQ2** [comment]
Hello, is it possible to send and read a CAN msg data ? (ussually I just can do it on Trace window)

**Reply by chaitu-ycr** [reply]
send/receive CAN message not possible. Vector didn't provide COM method to do this.

**Comment by anchithAcharya** [comment]
Hello, is it possible to attach a listener to a signal value instead of getting its value just once?

**Comment by Bkmaha** [comment]
Hello, is it possible to add a test unit from the Vtest studio executable file(.vtuexe)
